### ISSUE TITLE
render and renderFile Not Working

### ISSUE THREAD

**Issue by ebebbington** [syumai/dejs#40]
**Problem**

Returned value of using the `render` and `renderFile` method always returns `Buffer {}` and not the markup content.

**Expectation**

Returns the markup with the data within populated

**Steps to Reproduce**
```
// index.ejs
<body>
  <% if (name) { %>
    <h1>hello, <%= name %>!</h1>
  <% } %>
</body>
```

```
// app.ts
import { renderFile } from "https://deno.land/x/dejs/mod.ts";
const tpl = await renderFile(`${Deno.cwd()}/index.ejs`, { name: 'world'});
console.log(tpl) // Buffer {}
```

Has the same result when using the `render` method. Can clarify that it is correctly reading the contents of the file

**Comment by syumai** [comment]
@ebebbington Thanks! I'll check this later.

**Comment by syumai** [comment]
@ebebbington Would you provide information about your `Deno version` and `dejs version` ?

**Comment by ebebbington** [comment]
Deno version: 0.39.0
dejs Version: 0.3.5

**Comment by syumai** [comment]
@ebebbington 
It's expected behavior.
`dejs.renderFile()` returns `Deno.Reader`. Not `string`.
If you want result as `string`, please use Deno.Buffer or Deno.readAll & encode it into string.
Here's a example using Buffer.

```ts
import { renderFile } from 'https://deno.land/x/dejs@0.3.5/mod.ts';
const tpl = await renderFile(`${Deno.cwd()}/index.ejs`, { name: 'world' });
const buf = new Deno.Buffer();
await Deno.copy(buf, tpl);
console.log(buf.toString());
```

**Comment by ebebbington** [comment]
Oh right, thanks!:) I feel like this isn't immediently obvious in the docs, don't suppose i could requested it to be expanded upon if you feel the same? For example:

```
... // usual stuff
const buf = new Deno.buffer()
await Deno.copy(buf, tpl)
const rawHtml = buf.toString()
```

**Comment by dbohdan** [comment]
`Buffer#toString` no longer works in Deno v1.0.0-rc1 due to https://github.com/denoland/deno/pull/4992.  The fix is to decode the contents of the Buffer yourself:

```js
import { renderFile } from 'https://deno.land/x/dejs@0.4.0/mod.ts';
const tpl = await renderFile(`${Deno.cwd()}/index.ejs`, { name: 'world' });
const buf = new Deno.Buffer();
await Deno.copy(tpl, buf);
console.log(new TextDecoder().decode(buf.bytes());
```

**Comment by CkCreative** [comment]
I am trying to set up and this is the error I am getting.. `Deno 1.0.0`
```typescript
const { cwd, stdout, copy } = Deno;
import { renderFile } from 'https://deno.land/x/dejs/mod.ts';


(async () => {
  const output = await renderFile(`${cwd()}/views/index.ejs`, _);
  await copy(stdout, output);
})();
```
## The error...

``` javascript
const stdout: Deno.Writer & Deno.WriterSync & Deno.Closer & {
    rid: number;
}
Argument of type 'Writer & WriterSync & Closer & { rid: number; }' is not assignable to parameter of type 'Reader'.
  Property 'read' is missing in type 'Writer & WriterSync & Closer & { rid: number; }' but required in type 'Reader'.ts(2345)
lib.deno.d.ts(203, 5): 'read' is declared here.```

**Comment by syumai** [comment]
@CkCreative 
There was breaking change on `copy` function interface.
It became `copy(Reader, Writer)`, not `copy(Writer, Reader)`.
Please swap arguments order.
https://github.com/denoland/deno/pull/4885

**Comment by syumai** [comment]
I'll add `renderToString` and `renderFileToString` functions.

**Comment by syumai** [comment]
I've added these functions and released as `0.6.0`.
This will provide clear solutions for this issue.

- **`renderFileToString`**`(filePath: string, params: Params): Promise<string>`
  - renders from file, outputs string
- **`renderToString`**`(body: string, params: Params): Promise<string>`
  - renders from string, outputs string

**Comment by syumai** [comment]
I'm going to close this issue. If the problem is not resolved yet, please reopen.
